### PR TITLE
Add Tutorial Field

### PR DESCRIPTION
**Pull request by maj0-0**

### CODE REVIEW

**Comment by maj0-0** [comment]
Still working on this, will merge after test cases are done

File: src/main/java/seedu/address/model/person/Tutorial.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Student's tutorial in StudentConnect.
+ * Guarantees: immutable;
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/person/Tutorial.java:8]:
```suggestion
 * Guarantees: immutable; is valid as declared in {@link #isValidTutorial(String)}
```

File: src/main/java/seedu/address/model/person/Tutorial.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ */
+public class Tutorial {
+
+    public static final String MESSAGE_CONSTRAINTS = "Tutorials should be integers between 01 and 22, "
+                                                        + "separated by a comma.";
+
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/person/Tutorial.java:14]: insert this line of code
Constraint: Tutorial should only contain numbers, and it should be 2 digit long between 01 and 22
```suggestion
public static final String VALIDATION_REGEX = "^(0[1-9]|1\d|2[0-2])$";
```

File: src/main/java/seedu/address/model/person/Tutorial.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * Returns if a given string is a valid tutorial number between 01 and 22.
+     */
+    public static boolean isValidTutorial(String tutorial) {
+        try {
+            int tutorialNumber = Integer.parseInt(tutorial);
+            return tutorialNumber >= 1 && tutorialNumber <= 22;
+        } catch (NumberFormatException e) {
+            return false;
+        }
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/person/Tutorial.java:45]: change the method body to this
```suggestion
        return test.matches(VALIDATION_REGEX);
```

File: src/main/java/seedu/address/model/person/Tutorial.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Student's tutorial in StudentConnect.
+ * Guarantees: immutable;
+ */
+public class Tutorial {
+
+    public static final String MESSAGE_CONSTRAINTS = "Tutorials should be integers between 01 and 22, "
+                                                        + "separated by a comma.";
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/person/Tutorial.java:13]: shouldn't it be "separated by a space"?

File: src/main/java/seedu/address/model/person/Tutorial.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    public Tutorial(String tutorial) {
+        requireNonNull(tutorial);
+        checkArgument(isValidTutorial(tutorial), MESSAGE_CONSTRAINTS);
+        value = formatTutorial(tutorial);
+    }
+
+    /**
+     * Formats the tutorial to have leading zeros for numbers between 1 and 9 for consistency.
+     */
+    private String formatTutorial(String tutorial) {
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/person/Tutorial.java:31]: `VALIDATION_REGEX` forces valid tutorials to have a '0' before a single digit number, so I think this method may not be needed

**Review by PearlynnT — COMMENTED** [review]
Tutorial class needs to be edited

File: src/main/java/seedu/address/model/person/Tutorial.java [hunk 6]
```diff
@@ -5,12 +5,15 @@
...
- * Guarantees: immutable;
+ * Guarantees: immutable;  is valid as declared in {@link #isValidTutorial(String)}
  */
 public class Tutorial {
 
     public static final String MESSAGE_CONSTRAINTS = "Tutorials should be integers between 01 and 22, "
-                                                        + "separated by a comma.";
+                                                        + "separated by a space.\n"
+                                                        + "Constraint: Tutorial should only contain numbers, "
+                                                        + "and it should be 2 digit long between 01 and 22";
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/person/Tutorial.java:15]: This line is a repeat of line 12. Can you rephrase the message and follow how the other messages has been written for the other classes thanks!

Comment (maj0-0) [reply src/main/java/seedu/address/model/person/Tutorial.java]: Changed it to "Tutorials should be 2-digit numbers between 01 and 22, separated by spaces." Hope that's okay!

Comment (PearlynnT) [reply src/main/java/seedu/address/model/person/Tutorial.java]: Yup thank you!

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 7]
```diff
@@ -169,4 +172,17 @@ public void toStringMethod() {
         assertEquals(expected, editCommand.toString());
     }
 
+    @Test
```
Comment (wnchan) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:175]: LGTM! test case has been added for tutorial field.

**Review by wnchan — COMMENTED** [review]
LGTM! Test cases has been fixed.

File: src/test/java/seedu/address/model/person/TutorialTest.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+        // different types -> returns false
+        assertFalse(tutorial1.equals(5));
+
+        // null -> returns false
+        assertFalse(tutorial1.equals(null));
+
+        // different values -> returns false
+        assertFalse(tutorial1.equals(tutorial3));
+    }
```
Comment (PearlynnT) [inline src/test/java/seedu/address/model/person/TutorialTest.java:72]: Can help add a test case for `hashCode` thanks! Refer to the other test files
